assets/app/scripts/controllers/overview.js

```diff
@@ -384,7 +417,98 @@ angular.module('openshiftConsole')
       });
     });
 
+    /*
+     * The topology graph wants unique identifiers.
+     *
+     * One of the warts on the kubernetes/openshift API is that there
+     * is no unique global identifier per item. We construct them here from
+     * Kind/metadata.name ... which is good enough in a single namespace.
+     */
+    function makeId(name, kind) {
+      return kind + "/" + name;
+    }
+
+    var updateTimeout = null;
+
+    function updateTopology() {
+      updateTimeout = null;
+
+      var topologyRelations = [];
+      var topologyItems = { };
+
+      /*
+       * Put each of the items into the topology. Each of these
+       * are name -> object mappings.
+       */
+      [ $scope.pods,
+        $scope.routes,
+        $scope.services,
+        $scope.deployments,
+        $scope.deploymentConfigs
+      ].forEach(function(items) {
+        angular.forEach(items, function(item, name) {
```

Comment:
Good point. Will fix all introduced instances of variables called ```items``` or ```item``` and rename them.